test/builtin_dataset_mocks.py

```diff
@@ -72,6 +71,24 @@ def prepare(self, home, config):
         available_file_names = {path.name for path in root.glob("*")}
         required_file_names = {resource.file_name for resource in self.dataset.resources(config)}
         missing_file_names = required_file_names - available_file_names
+        extra_file_names = available_file_names - required_file_names
+
+        # Some datasets need to provide already preprocessed data, for example if the preprocessing includes downloads.
+        # Such data will be included in the `extra_file_names` while the raw data will be in the `missing_file_names`.
+        # This detects these cases and removes the corresponding entries from the sets since the files are neither extra
+        # nor missing.
+        if missing_file_names:
+            for missing in missing_file_names.copy():
+                extra_candidate = missing.split(".", 1)[0]
```

Comment:
Isn't this equivalent to ?

```suggestion
                extra_candidate = missing.split(".")[0]
```

Reply:
Yes. Just a force of habit to not split more than we need. Without the `1`, split will process the whole string whereas otherwise it would stop after the first occurrence. Since we only select first part anyway, there is no need to go to the full string. Not going to fight this if you insist.

Reply:
I had to google what this extra argument was. I'm not sure we're getting any perf boost for this, so this mostly just adds noise IMO. If we keep it, please use keyword args.